app/controllers/admins_controller.rb

```diff
@@ -5,12 +5,14 @@ def index
     @users = User.all
   end
 
-  # GET /admin_accept_user_invitation?invitation_token=abcdef
+  # GET /admin_accept_user_invitation?user_id=1
   def accept_user_invitation
     if !request.patch?
-      @user = User.find_by_invitation_token(params[:invitation_token], false)
+      @user = User.find(params[:user_id])

```

Comment:
Is this only used by the admin page?
In other words does the normal user pattern of accepting an email based invitation go through a different code path?

Reply:
Yes - the token that is used is a short hash of the token in the db and that works.  What wasn't working is using the token in the db and not the hash of it when accepting an invitation which is why I changed it to be by user id since this is only usable by the admins.

Reply:
You can see in another one of the tests that I had to change the invitation_token to the raw token generated and not stored to get the user invite acceptance to work within the user.steps system.

Reply:
Could you add a comment above accept_user_invitation indicating this is only used by admins, and that the real user invitation accepting happens elsewhere.

Also is this method locked down from regular users? With the previous invitation token approach it didn't really matter if regular users could access it since they wouldn't have the token.

Reply:
Sure - yes the admin controller is locked down:

```
class AdminsController < ApplicationController
  before_filter :authenticate_admin!
```

authenticate_admin! is a devise method.